feat(core): mark input, output and model APIs as stable

pkozlowski-opensource opened the pull request:
This commit marks the input, output and model APIs as stable (along with the associated APIs) and thus exits the dev preview phase for those APIs.

Documentation updates will follow.

JeanMeche commented on packages/core/src/authoring/input/input_signal.ts:
```diff
@@ -14,8 +14,6 @@ import {Signal} from '../../render3/reactivity/api';
 import {INPUT_SIGNAL_NODE, InputSignalNode, REQUIRED_UNSET_VALUE} from './input_signal_node';
 
 /**
- * @developerPreview
```
Shouldn't it be replaced by `@publicAPI` ?

pkozlowski-opensource replied:
They absolutely should :-) Thnx.

pkozlowski-opensource committed:
feat(core): mark input, output and model APIs as stable

This commit marks the input, output and model APIs as stable
(along with the associated APIs) and thus exits the dev preview
phase for those APIs.

devversion commented:
🎉🚀

bencodezen reviewed (approved):
This is great. Thank you @pkozlowski-opensource!

NicolasVargas-Hilti commented:
Hi @pkozlowski-opensource, is this retro-applied to Angular 18 or is Angular 19 shipped with other core developments that make these APIs only stable there?

JeanMeche commented:
@NicolasVargas-Hilti Stability is mostly a maintenance status. When an API is stable, breaking changes only land in major versions and APIs can be removed only after 2 major versions of deprecations.
